handshake_client.go

```diff
@@ -435,23 +435,26 @@ func (hs *clientHandshakeState) doFullHandshake() error {
 			return fmt.Errorf("tls: client certificate private key of type %T does not implement crypto.Signer", chainToSend.PrivateKey)
 		}
 
-		var signatureType uint8
+		var signatureType signatureType
 		switch key.Public().(type) {
 		case *ecdsa.PublicKey:
 			signatureType = signatureECDSA
 		case *rsa.PublicKey:
-			signatureType = signatureRSA
+			signatureType = signatureRSAPKCS1
```

Comment:
so now this value will be 0? I have a feeling something is not quite right

Reply:
It is not sent over the network, so the exact value does not matter. Maybe I should use `iota + 1` though to allow for "uninitialized" values to be recognized.

Reply:
That could work, or we just assign 0 to some `None` constant.